### ISSUE TITLE
Stringy enum JSON serialization/deserialization

### ISSUE THREAD

**Issue by BenRKarl** [googleapis/proto-plus-python#52]
There appears to be a difference in behavior between wrapped protobufs and raw protobufs loaded directly from a pb2 file.

Serializing a raw protobuf message instance from a pb2 file include the Enum field name in the JSON, i.e.:

`'{\n  "campaign": {\n    "status": "PAUSED"\n  }\n}'`

While serializing a wrapped protobuf message includes the field number, i.e.:

`'{\n  "campaign": {\n    "status": 3\n  }\n}'`

Similar issue going in the deserialization direction. When deserializing JSON that includes the field name, this error is raised:

```
    raise ParseError('Failed to parse {0} field: {1}.'.format(name, e))
google.protobuf.json_format.ParseError: Failed to parse campaign field: Failed to parse create field: Failed to parse status field: invalid literal for int() with base 10: 'PAUSED'...
```

From [this line in json_format](https://github.com/protocolbuffers/protobuf/blob/master/python/google/protobuf/json_format.py#L592).

**Comment by BenRKarl** [comment]
In the above serializing is done the same as in this Issue: https://github.com/googleapis/proto-plus-python/issues/37

`json = MessageToJson(campaign._pb)`

The raw protos are serialized similarly:

`json = MessageToJson(campaign)`

**Comment by software-dov** [comment]
I'm having difficulty reproducing the error. All attempts at creating proto messages with enum fields aggressively turn the field into an integer representation when serialized to json. Can you direct me to the environment you're using, or give me some tips in setting up the minimum steps for reproduction?

**Comment by nwbirnie** [comment]
I thought the serialization which triggered this bug was the Java protobuf implementation? In which case it would be `JsonFormat.printer().print(message)`.

In other words, I don't think that this is a serialization issue, rather it's a case of the deserialization being unable to read enums from string.

**Comment by BenRKarl** [comment]
I can work on a repro soon, meanwhile a bit more context - 

Our benchmarks read mutate operations from JSON, which currently has the enum field name (i.e. `"PAUSED'`). That was fine for the monolithic codebase to deserialize, however the microgenerator interface now generates the above error.

Serialization isn't a blocking issue for us, but there's a difference in the output when using the same `MessageToJson` helper function, which seems potentially problematic.

**Comment by software-dov** [comment]
Update: internally, the structures that are being traversed in order to send messages to json describe all enums as INT32. I'm now in the process of determining what all needs to change in order for it to think of them as enums and to know what names are permitted enum variants. I don't have a good idea yet of how much work is involved with that.

**Comment by BenRKarl** [comment]
Thanks Dov - sorry am late with this, but here's a simply repro case:

Start with this `foo.proto` file:

```
syntax = "proto3";

enum Bar {
  CAT = 0;
  DOG = 1;
}

message Foo {
  Bar status = 1;
}
```

Run it through `protoc`

`protoc --python_out=. foo.proto`

Then in a Python repl:

```
>>> from foo_pb2 import Foo
>>> foo = Foo(status=1)
>>> print(foo)
status: DOG
>>> from google.protobuf.json_format import MessageToJson
>>> MessageToJson(foo)
'{\n  "status": "DOG"\n}' # here a proto-plus-wrapped proto would be '{\n  "status": 1\n}'
```

**Comment by software-dov** [comment]
Thanks, that gives me plenty to work with.
I think that this may be a non-trivial change in proto-plus. I will continue to work on it, but I don't believe that I can finish by the end of the week.

**Comment by BenRKarl** [comment]
Sounds good keep me posted and let me know if there's anything I can do to help.

**Comment by nwbirnie** [comment]
@software-dov @BenRKarl hey folks, just wanted to check in to see if there's any progress on this?

**Comment by software-dov** [comment]
This issue has been sidelined for the past few days. We will be getting to as a top priority for the next sprint.

**Comment by software-dov** [comment]
Fixed by #107
